html/user/submit_rpc_handler.php

```diff
@@ -588,9 +603,9 @@ function get_batch($r) {
         $batch_name = BoincDb::escape_string($batch_name);
         $batch = BoincBatch::lookup_name($batch_name);
     } else {
-        xml_error(-1, "BOINC server: batch not specified");
+        xml_error(-1, "batch not specified");
     }
-    if (!$batch) xml_error(-1, "BOINC server: no such batch");
+    if (!$batch) xml_error(-1, "no such batch");
```

Comment:
Scrutinizer found an issue about the scope of `$batch`. It does not know that `xml_error()` terminates execution so it found an execution path where `$batch` is not defined at this point. This can be solved by adding `$batch = NULL;` before the if block.